src/algorand/AlgorandEngine.ts

```diff
@@ -95,6 +106,152 @@ export class AlgorandEngine extends CurrencyEngine<
     }
     this.minimumAddressBalance = this.networkInfo.minimumAddressBalance
     this.totalReserve = this.minimumAddressBalance
+
+    this.otherMethods = {
+      // Wallet Connect utils
+      wcInit: async (
+        wcProps: WcProps,
+        walletName: string = 'Edge'
+      ): Promise<WcDappDetails> => {
+        return await timeout(
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          new Promise((resolve, reject) => {
+            const connector = new WalletConnect({
+              uri: wcProps.uri,
+              storageId: wcProps.uri,
+              clientMeta: {
+                description: 'Edge Wallet',
+                url: 'https://www.edge.app',
+                icons: ['https://content.edge.app/Edge_logo_Icon.png'],
+                name: walletName
+              }
+            })
+
+            connector.on(
+              'session_request',
+              (error: Error | null, payload: AlgoWcRpcPayload) => {
+                if (error != null) {
+                  this.error(`Wallet connect session_request`, error)
+                  throw error
+                }
+
+                const params = asWcSessionRequestParams(payload).params[0]
+                const dApp = { ...params, timeConnected: Date.now() / 1000 }
+
+                // Set connector in memory
+                this.tools.walletConnectors[wcProps.uri] = {
+                  connector,
+                  wcProps,
+                  dApp
+                }
+                resolve(dApp)
+              }
+            )
+
+            // Subscribe to call requests
+            connector.on(
+              'call_request',
+              (error: Error | null, payload: AlgoWcRpcPayload) => {
+                try {
+                  if (error != null) throw error
+                  const cleanPayload = asAlgorandWalletConnectPayload(payload)
+                  const params = cleanPayload.params[0][0]
+                  const algoTx = decodeUnsignedTransaction(
+                    base64.parse(params.txn)
+                  )
+
+                  const nativeAmount =
+                    algoTx.amount != null ? algoTx.amount.toString() : '0'
+                  const networkFee = algoTx.fee.toString()
```

Comment:
Would `toFixed` make more sense? If the number is really big, you might get "1e+30", which wouldn't be valid for biggystring.